src/go/k8s/cli/configurator/main.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2021 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+// main package. This line should be remove after merging the following
+// https://github.com/vectorizedio/redpanda/pull/753
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/spf13/afero"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/config"
+	"gopkg.in/yaml.v3"
+)
+
+var (
+	hostName		string
+	svcFQDN			string
+	configSourceDir		string
+	configDestination	string
+	redpandaRPCPort		int
+)
+
+var errorMissingEnvironmentVariable = errors.New("missing environment variable")
+
+func main() {
+	if err := checkEnvVars(); err != nil {
+		log.Fatalf("%s", fmt.Errorf("unable to get the environment variables: %w", err))
+	}
+
+	fs := afero.NewOsFs()
+	v := config.InitViper(fs)
+	v.AddConfigPath(configSourceDir)
+
+	if err := v.ReadInConfig(); err != nil {
+		log.Fatalf("%s", fmt.Errorf("unable to read the redpanda configuration file: %w", err))
+	}
+
+	cfg := &config.Config{}
+	decoderConfig := mapstructure.DecoderConfig{
+		Result:	cfg,
+		// Sometimes viper will save int values as strings (i.e.
+		// through BindPFlag) so we have to allow mapstructure
+		// to cast them.
+		WeaklyTypedInput:	true,
+	}
+
+	decoder, err := mapstructure.NewDecoder(&decoderConfig)
+	if err != nil {
+		log.Fatalf("%s", fmt.Errorf("unable to create decoder config: %w", err))
+	}
+
+	err = decoder.Decode(v.AllSettings())
+	if err != nil {
+		log.Fatalf("%s", fmt.Errorf("unable to decode: %w", err))
+	}
+
+	log.Printf("Decode done")
+
+	hostIndex, err := hostIndex(hostName)
```

Comment:
I think we need some docs here, I don't really understand what's host index and why we need it

Reply:
I will create godoc to describe why we use index.

FYI we didn't have docs in the bash script
REF https://github.com/vectorizedio/redpanda/pull/755/files#diff-74fd28b7b800bebf72d52c4b3beaab538ca2605ce41735a05fcd6b4fd9b3963fL109